Libraries/D3D12RaytracingFallback/src/FallbackLayerUnitTests/fallbacklayerunittests.cpp

```diff
@@ -1003,6 +1045,357 @@ namespace FallbackLayerUnitTests
+
+            D3D12_GET_RAYTRACING_ACCELERATION_STRUCTURE_PREBUILD_INFO_DESC bottomLevelDesc = {};
+            bottomLevelDesc.Type = D3D12_RAYTRACING_ACCELERATION_STRUCTURE_TYPE_BOTTOM_LEVEL;
+            bottomLevelDesc.Flags = D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAG_PREFER_FAST_BUILD;
+            bottomLevelDesc.NumDescs = 1;
+            bottomLevelDesc.pGeometryDescs = &gpuGeomDesc;
+            bottomLevelDesc.DescsLayout = D3D12_ELEMENTS_LAYOUT_ARRAY;
+
+            D3D12_RAYTRACING_ACCELERATION_STRUCTURE_PREBUILD_INFO infoNoUpdate = {};
+            pBuilder->GetRaytracingAccelerationStructurePrebuildInfo(&bottomLevelDesc, &infoNoUpdate);
+
+            // Turn on updates, make sure the memory gets allocated.
+            bottomLevelDesc.Flags |= D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAG_ALLOW_UPDATE;
+
+            D3D12_RAYTRACING_ACCELERATION_STRUCTURE_PREBUILD_INFO infoWithUpdate = {};
+            pBuilder->GetRaytracingAccelerationStructurePrebuildInfo(&bottomLevelDesc, &infoWithUpdate);
+
+            Assert::AreEqual(infoNoUpdate.ScratchDataSizeInBytes, infoWithUpdate.ScratchDataSizeInBytes, L"Allowing update allocated scratch data.");
+            Assert::IsTrue(infoWithUpdate.ResultDataMaxSizeInBytes == infoNoUpdate.ResultDataMaxSizeInBytes + updateExtraDataSize, L"Data allocated for update doesn't match expected.");
+        }
+
+        TEST_METHOD(StoreSortResultForUpdate) {
+            ID3D12Device &device = m_d3d12Context.GetDevice();
+            std::unique_ptr<FallbackLayer::IAccelerationStructureBuilder> pBuilder =
+                std::unique_ptr<FallbackLayer::IAccelerationStructureBuilder>(
+                    new FallbackLayer::GpuBvh2Builder(&device, m_d3d12Context.GetTotalLaneCount(), 0));
+            InternalFallbackBuilder builderWrapper(pBuilder.get());
+
+            UINT numVertices = VERTEX_COUNT(ReferenceVerticies1);
+            UINT numTriangles = numVertices / 3;
+
+            CpuGeometryDescriptor geomDesc = CpuGeometryDescriptor(ReferenceVerticies1, VERTEX_COUNT(ReferenceVerticies1));
+
+            std::unique_ptr<BYTE[]> pData;
+            BuildBottomLevelAccelerationStructureAndGetCpuData(
+                builderWrapper,
+                &geomDesc,
+                1,
+                pData,
+                D3D12_ELEMENTS_LAYOUT_ARRAY,
+                D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAG_PREFER_FAST_BUILD | D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAG_ALLOW_UPDATE);
+            
+            const BYTE *pOutputBVH = pData.get();
+            BVHOffsets offsets = *(BVHOffsets*)pOutputBVH;
+            AABBNode *pNodeArray = (AABBNode*)((BYTE *)pOutputBVH + offsets.offsetToBoxes);
+            Primitive *pPrimitiveArray = (Primitive*)((BYTE *)pOutputBVH + offsets.offsetToVertices);
+            UINT *pSortResults = (UINT *)((BYTE *)pOutputBVH + offsets.totalSize);
+
+            for (UINT i = 0; i < numTriangles; i++) {
```

Comment:
I'm a little worried about this just because it's assuming that ReferenceVertices1 is already sorted (which it coincidentally is). Also it's not clear to me what this is testing that isn't already hit by the above tests that run it through the GPU validator